dbms/src/Storages/DeltaMerge/File/DMFileReader.h

```diff
@@ -142,7 +154,7 @@ class DMFileReader
     /// Filters
     DMFilePackFilter pack_filter;
 
-    std::vector<size_t> skip_packs_by_column{};
+    std::vector<bool> must_seek;
```

Comment:
avoid using `std::vector<bool>`

Reply:
`DMFileReader` will be created for each segment in a query. If multiple queries are running at the same time, a lot of `DMFileReader` will be created. And if there are cop tasks, it is even worse. So reducing memory is important for `DMFileReader`.

And since `must_seek` is a private member, which means it only can be accessed inside, so no outer reference refers to it, meanwhile, the life cycle of the inside reference will always shorter than `must_seek` itself. So it is safe.

Reply:
Why not use bitset?

Reply:
`std::bitset` can not decide size in run time. `std::vecotr<bool>` is safe in this case.

Reply:
How about `boost::dynamic_bitset`? It is dynamic. Moreover, it supports efficient batch set bits (while `std::fill(std::vector<bool>)` does not stipulate its complexity and may have poor performance in some stds)

Reply:
`boost::dynamic_bitset::operator[]` also return a reference.

Anyway, `must_seek` is removed.